src/TBuffer.cpp

```diff
@@ -787,10 +791,24 @@ void TBuffer::setBufferSize(int s, int batch)
     if (batch >= s) {
         batch = s / 10;
     }
-    mLinesLimit = s;
+    // clip the maximum to something reasonable, else users will abuse this, and then complain
+    mLinesLimit = std::min(s, getMaxBufferSize());
```

Comment:
Maybe log a warning somewhere (stderr?) that the buffer size was made smaller than requested?

Reply:
stderr sounds good to me.

Reply:
Apart from Windows maybe? `qWarning()` might be an option...

Reply:
Yeah, that's the one I was thinking of.

Reply:
Fixed.